fluent/toInterfaceValue.go

```diff
@@ -0,0 +1,67 @@
+package fluent
+
+import (
+	"github.com/ipld/go-ipld-prime/datamodel"
+)
+
+func ToInterfaceValue(node datamodel.Node) (interface{}, error) {
+	switch k := node.Kind(); k {
+	case datamodel.Kind_Invalid:
+		panic("invalid node")
+	case datamodel.Kind_Null:
+		return nil, nil
+	case datamodel.Kind_Bool:
+		return node.AsBool()
+	case datamodel.Kind_Int:
+		return node.AsInt()
+	case datamodel.Kind_Float:
+		return node.AsFloat()
+	case datamodel.Kind_String:
+		return node.AsString()
+	case datamodel.Kind_Bytes:
+		return node.AsBytes()
+	case datamodel.Kind_Link:
+		return node.AsLink()
+	case datamodel.Kind_Map:
+		outMap := make(map[string]interface{})
+		for mi := node.MapIterator(); !mi.Done(); {
+			k, v, err := mi.Next()
+			if err != nil {
+				return nil, err
+			}
+			kVal, err := k.AsString()
```

Comment:
perhaps leave a comment explaining that, for now, we assume all keys are strings.

it's an okay assumption to make right now, but we should also make it clear.

Reply:
My understanding is that IPLD maps always have string keys. It says this in the docs:
https://ipld.io/docs/intro/primer/#finer-points
> Map keys are strings.

Reply:
Ah, you're right, the docs even say that!